Sources/Kitura/CodableRouter.swift

```diff
@@ -600,21 +603,37 @@ extension Router {
         }
     }
 
-    internal func parameterIsPresent(in route: String) -> Bool {
-        if route.contains(":") {
-            let paramaterString = route.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
-            let parameter = paramaterString.count > 0 ? paramaterString[1] : ""
-            Log.error("Erroneous path '\(route)', parameter ':\(parameter)' is not allowed. Codable routes do not allow parameters.")
+    /// Precondition: Path is known to contain a : character
+    func pathHasSingleParamIdAsSuffix(_ path: String) -> Bool {
+        let paramaterString = path.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
+        let parameter = paramaterString.count > 0 ? paramaterString[1] : ""
+        return parameter.isEmpty || parameter.elementsEqual("id")
+    }
+
+    func pathSyntaxIsValid(_ path: String, identifierExpected: Bool) -> Bool {
+        let identifierSupplied = path.contains(":")
+        if !identifierExpected && identifierSupplied {
```

Comment:
Would this be clearer if we did

```swift
switch (identifierExpected, identifierSupplied) {
    ...
}
```

and pattern-matched?

Reply:
@ianpartridge   The above code would look like this if we change it to switch-case 
```
switch(identifierExpected, identifierSupplied) {
        case(false, true) :
            Log.error("Path '\(path)' is not allowed: Codable routes do not allow path parameters.")
            return false
        case (true, false):
            Log.warning("Identifier expected for '\(path)'. :id will be automatically appended to the path.")
            return true
        case (true, true):
            guard pathHasSingleParamIdAsSuffix(path) else {
            Log.error("Erroneous path '\(path)' is not allowed. Codable routes support a trailing id parameter only.")
            return false
            }
            return true
        case (false, false):
            return true
        }
```
However, for two reasons I was thinking not to go for it is , there is no logical reason behind  coupling```identifierExpected``` and  ```identifierSupplied ``` as a tuple. Also, I believe  the readability is not better than the legacy if-else approach. Please let me know which approach we need to take ..
@djones6 please let us know your opinion too ..

Reply:
Agreed, please make this change. Put the common `(false, false)` fast-path as the first case.